packages/flutter_reactive_ble/lib/src/device_connector.dart

```diff
@@ -85,7 +85,17 @@ class DeviceConnectorImpl implements DeviceConnector {
     Map<Uuid, List<Uuid>>? servicesWithCharacteristicsToDiscover,
     Duration? connectionTimeout,
   }) {
-    if (_deviceScanner.currentScan != null) {
+    if (_deviceIsDiscoveredRecently(
```

Comment:
why don't you just stop the scan and connect with `connectToAdvertisingDevice ` ?